Source/DafnyCore/Pipeline/Compilation.cs

```diff
@@ -356,15 +356,27 @@ private async Task VerifyUnverifiedSymbol(bool onlyPrepareVerificationForGutterT
         return result;
       });
       if (updated || randomSeed != null) {
-        updates.OnNext(new CanVerifyPartsIdentified(canVerify,
-          tasksPerVerifiable[canVerify].ToList()));
+        updates.OnNext(new CanVerifyPartsIdentified(canVerify, tasks));
       }
 
       // When multiple calls to VerifyUnverifiedSymbol are made, the order in which they pass this await matches the call order.
       await ticket;
 
       if (!onlyPrepareVerificationForGutterTests) {
+        foreach (var tokenTasks in tasks.GroupBy(t =>
+                  BoogieGenerator.ToDafnyToken(true, t.Token)).
+                  OrderBy(g => g.Key)) {
+          var functions = tokenTasks.SelectMany(t => t.Split.HiddenFunctions.Select(f => f.tok).
+            OfType<FromDafnyNode>().Select(n => n.Node).
```

Comment:
Wrapping tokens to filter functions is not ideal. There are multi competing token wrapping strategies that will end up failing this test in the future. For example, refinement tokens will wrap your FromDafnyNode token making them not pass this test. There are other wrapped tokens types, like auto-generated tokens, nested tokens, override tokens for traits, etc.
I have always advocated for [replacing tokens with a single meta-data field](https://github.com/dafny-lang/dafny/issues/3354) on which we would add some flags as needed.

I'm not expecting you to do this refactoring, so in the meantime, please use something like `FromDafnyNode.Has()` that would recursively traverse the tokens to find one of its kind.

Reply:
Here the wrapping happens right before handing the token to Boogie, so Dafny won't do any further wrapper, and Boogie might do some wrapping on tokens but only use that for nodes that it creates, so not on the nodes that I'm interested in. I don't think there's a potential issue here. If there is one, how can we test for it?

Reply:
Ok, good to know that the risk is mitigated. I think that we still have the risk that later, someone will wrap your token after your own wrapping and their test cases will be orthogonal to yours, so both won't be tested together.
If you want to make sure your wrapping is the last one before sending the token to boogie, I'd suggest: changing the name to something like `DafnyMetaData` so that future development will become compelled to add metadata here rather than adding a new wrapping token.